WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasTwoColumnDropDown.lua

```diff
@@ -0,0 +1,210 @@
+local Type, Version = "WeakAurasTwoColumnDropdown", 1
+local AceGUI = LibStub and LibStub("AceGUI-3.0", true)
+if not AceGUI or (AceGUI:GetWidgetVersion(Type) or 0) >= Version then return end
+
+local function errorhandler(err)
+  return geterrorhandler()(err)
+end
+
+local function safecall(func, ...)
+  if func then
+    return xpcall(func, errorhandler, ...)
+  end
+end
+
+AceGUI:RegisterLayout("TwoColumn",
+  function(content, children)
+    local height = 0
+    local width = content.width or content:GetWidth() or 0
+    for i = 1, #children do
+      local child = children[i]
+
+      local frame = child.frame
+      frame:ClearAllPoints()
+      if child.hideMe then
+        frame:Hide()
+      else
+        frame:Show()
+      end
+      if i == 1 then
+        frame:SetPoint("TOPLEFT", content)
+      else
+        frame:SetPoint("TOPLEFT", children[i-1].frame, "TOPRIGHT")
+      end
+
+      if child.width == "relative" then
+        child:SetWidth(width * child.relWidth)
+
+        if child.DoLayout then
+          child:DoLayout()
+        end
+      end
+
+      height = max(height, frame.height or frame:GetHeight() or 0)
+    end
+    safecall(content.obj.LayoutFinished, content.obj, nil, height)
+  end)
+
+local methods = {
+  ["OnAcquire"] = function(widget)
+    local firstDropdown = AceGUI:Create("Dropdown")
+    local secondDropDown = AceGUI:Create("Dropdown")
+
+    firstDropdown:SetRelativeWidth(0.5)
+    firstDropdown:SetPulloutWidth(200)
+    secondDropDown:SetRelativeWidth(0.5)
+    secondDropDown:SetLabel(" ")
+    secondDropDown:SetPulloutWidth(200)
+
+    widget.firstDropdown = firstDropdown
+    widget.secondDropDown = secondDropDown
+
+    widget:SetLayout("TwoColumn")
+    widget.SetLayout = function()
+      -- AceGui wants to set a default layout, but we don't want that
+    end
+
+    widget:AddChild(firstDropdown)
+    widget:AddChild(secondDropDown)
+
+    local OnFirstDropdownValueChanged = function(self, event, value)
+      local displayName = widget.userdata.firstList[value]
+      local treeValue = widget.userdata.tree[displayName]
+      if type(treeValue) == "table" then
+        local secondList = {}
+        for displayName, _ in pairs(treeValue) do
```

Comment:
You could simplify this to `for displayName in pairs(treeValue) do` if you like.